rpm: put numpy and scipy deps in distro conditionals

smithfarm opened the pull request:
Without this patch, attempts to install the ceph-mgr-diskprediction-local RPM
fail on SUSE platforms with the following error:

```
can't install ceph-mgr-diskprediction-local-14.1.0.402+ga396e8bf3b-3742.1.noarch:
  nothing provides numpy needed by ceph-mgr-diskprediction-local-14.1.0.402+ga396e8bf3b-3742.1.noarch
  nothing provides scipy needed by ceph-mgr-diskprediction-local-14.1.0.402+ga396e8bf3b-3742.1.noarch
```

Also take into account package naming differences between Fedora and
RHEL/CentOS.

Signed-off-by: Kefu Chai <kchai@redhat.com>
Signed-off-by: Nathan Cutler <ncutler@suse.com>

tchaikov commented on ceph.spec.in:
```diff
@@ -488,8 +488,14 @@ BuildArch:      noarch
 Requires:       numpy
 Requires:       scipy
+%endif
+%if 0%{?suse_version}
```
yeah, this is more correct! but i think we can do better, like:

```
%if 0{?fedora} || 0%{?suse_version}
Requires:       python%{_python_buildid}-numpy
%if 0{without python2}
Requires:       python3-scipy
%else
Requires:       python2-scipy
%endif
%endif
%if 0%{?rhel}
Requires:       numpy
Requires:       scipy
%endif
```

see https://rpmfind.net/linux/rpm2html/search.php?query=python2-scipy&submit=Search+...&system=&arch=
and https://rpmfind.net/linux/rpm2html/search.php?query=python-scipy&submit=Search+...&system=&arch=

smithfarm commented:
@tchaikov OK, please take another look (I also updated the commit message and the PR description).

dmick commented:
Sure?...

smithfarm commented:
@dmick I am not sure this is correct, but @tchaikov posted the following:

see https://rpmfind.net/linux/rpm2html/search.php?query=python2-scipy&submit=Search+...&system=&arch=
and https://rpmfind.net/linux/rpm2html/search.php?query=python-scipy&submit=Search+...&system=&arch=
and https://packagehub.suse.com/packages/python-scipy/

ktdreyer commented on ceph.spec.in:
```diff
@@ -488,8 +488,18 @@ BuildArch:      noarch
 Group:          System/Filesystems
 %endif
 Requires:       ceph-mgr = %{_epoch_prefix}%{version}-%{release}
+%if 0{?fedora} || 0%{?suse_version}
```
Looks like this might be a syntax error here, "%if 0{?fedora}"

We need this to be `%if 0{?fedora}"

And we also need RHEL 8, so `%if 0{?fedora} || 0%{?rhel} > 7 || 0%{?suse_version}`

liewegas commented:
i think this breaks teh build, see https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos7,DIST=centos7,MACHINE_SIZE=huge/18418//consoleFull

smithfarm commented:
@liewegas Sorry for breaking the build and thanks @ktdreyer for finding the syntax error. Fixed now, rebased, and building at https://shaman.ceph.com/builds/ceph/wip-rpm-numpy-scipy/edafc80cd1b63f8608743864e3b98915ae01460e/

tserong commented on ceph.spec.in:
```diff
@@ -493,8 +493,18 @@ BuildArch:      noarch
 Requires:       ceph-mgr = %{_epoch_prefix}%{version}-%{release}
+%if 0%{?fedora} || 0%{?rhel} > 7 || 0%{?suse_version}
+Requires:       python%{_python_buildid}-numpy
+%if 0{without python2}
```
Does this also need an extra '%', i.e. `%if 0%{without python2}`?

Also, why not just `Requires: python%{_python_buildid}-scipy` as is done with numpy?

tchaikov replied:
`_python_buildid` expands to an empty string if `0%{with python2}`. but we need "2" in this case.

tserong replied:
Ah, OK.

tchaikov commented:
s/pit/out/ in my last comment.

smithfarm commented:
Thanks. Fixed.

tchaikov commented:
yeah, i feel that RHEL/fedora are trying to remove "python" from these distros.

ktdreyer commented:
That's right
